onnxruntime/core/session/inference_session.h

```diff
@@ -403,6 +403,10 @@ class InferenceSession {
                             int32_t partial_graph_index);
 #endif
 
+#ifdef ENABLE_TRAINING_APIS
+  std::shared_ptr<onnxruntime::Model> GetModel() noexcept { return model_; };
+#endif
```

Comment:
Can we access this by using a derived class in the training code instead of by modifying the InferenceSession API? 

Also, in general, unless we're transferring/extending ownership we shouldn't expose that the model is stored in a shared_ptr (or unique_ptr) and could just return a `Model&`.